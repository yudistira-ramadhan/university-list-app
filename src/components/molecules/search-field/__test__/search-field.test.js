import React from 'react'
import ReactDOM from 'react-dom'
import SearchField from '../search-field'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(<SearchField />, div)
})

it('renders search field correctly', () => {
  const { getByTestId, getByPlaceholderText } = render(<SearchField />)

  expect(getByTestId('search-button')).toBeTruthy()

  expect(getByPlaceholderText('default placeholder')).toBeTruthy()
})

describe('input value', ()=> {
  it('updates on change', ()=> {

    const { getByPlaceholderText } = render(<SearchField/>)

    const searchInput = getByPlaceholderText('default placeholder')

    fireEvent.change(searchInput, { target: { value: 'test' } })

    expect(searchInput.value).toBe('test')
  })
})

describe('search button', ()=>{
  describe('with empty value', ()=> {
    it('does not trigger request search function', ()=> {
      const onHandleClick = jest.fn()

      const { getByTestId } = render(<SearchField onClick={ onHandleClick } />)

      fireEvent.click(getByTestId('search-button'))

      expect(onHandleClick).not.toHaveBeenCalled()
    })
  })

  describe('with value inside field', ()=>{
    it('triggers onHandleChange function', ()=>{
      const onHandleChange = jest.fn()

      const { getByTestId, getByPlaceholderText } = render(<SearchField onHandleChange={ onHandleChange } />)

      const searchInput = getByPlaceholderText('default placeholder')

      fireEvent.change(searchInput, { target: { value: 'test' } })

      fireEvent.click(getByTestId('search-button'))

      expect(onHandleChange).toHaveBeenCalled()
    })
  })
})

it('matches snapshot', () => {
  const tree = renderer.create(<SearchField />).toJSON()
  expect(tree).toMatchSnapshot()
})