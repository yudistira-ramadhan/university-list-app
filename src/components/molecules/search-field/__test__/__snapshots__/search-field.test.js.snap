// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="d-flex"
>
  <input
    className="mr-2 form-control"
    name="default name"
    onChange={[Function]}
    placeholder="default placeholder"
    type="text"
  />
  <button
    aria-label={null}
    className="btn btn-outline-primary"
    data-testid="search-button"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
